### PR TITLE
WindowServer: Set no active windows on a click outside of any window

### PR DESCRIPTION
**Pull request by shannonbooth**

### CODE REVIEW

File: Servers/WindowServer/WSWindowManager.cpp [hunk 1]
```diff
@@ -905,11 +905,14 @@ void WSWindowManager::process_mouse_event(WSMouseEvent& event, WSWindow*& hovere
             return IterationDecision::Continue;
         });
     } else {
+        bool mouse_event_in_window = false;
```
Comment (awesomekling) [inline Servers/WindowServer/WSWindowManager.cpp]: Hmm, isn't this the same as the `hovered_window` variable?

File: Servers/WindowServer/WSWindowManager.cpp [hunk 2]
```diff
@@ -961,6 +964,10 @@ void WSWindowManager::process_mouse_event(WSMouseEvent& event, WSWindow*& hovere
             event_window_with_frame = &window;
             return IterationDecision::Break;
         });
+
+        // Clicked outside of any window
+        if (!mouse_event_in_window && event.type() == WSEvent::MouseDown)
+            set_no_active_window();
```
Comment (awesomekling) [inline Servers/WindowServer/WSWindowManager.cpp]:
```suggestion
            set_active_window(nullptr);
```

File: Servers/WindowServer/WSWindowManager.cpp (hunk 1 above)
Comment (shannonbooth) [reply Servers/WindowServer/WSWindowManager.cpp]: Oh whoops! It is too! Thanks :)

File: Servers/WindowServer/WSWindowManager.cpp (hunk 2 above)
Comment (shannonbooth) [reply Servers/WindowServer/WSWindowManager.cpp]: Ah okay - tried that at first but it crashed stuff. Must have just been a bug with the function! Fixed now though :)

**Pull request merged**